src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -673,7 +682,19 @@ private void parseMeta(Element e) {
     }
 
     //for ease of parsing, we assume a model comes before the controls, which isn't necessarily mandated by the xforms spec
-    private void parseModel(Element e) {
+    private void parseModel(Element e) throws XFormParseException {
+        modelAttributeProcessors.stream().forEach(processor -> {
+            for (int i = 0; i < e.getAttributeCount(); i++) {
```

Comment:
I'm always terrified to see any kind of nested iteration but I guess we don't really care too much about form parse time (within reason) and we generally expect at most a handful of attributes and processors.